html/browsers/the-window-object/proxy-getOwnPropertyDescriptor.html

```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>getOwnPropertyDescriptor() is correct for Proxy with host object target</title>
+  <link rel="help" href="https://html.spec.whatwg.org/multipage/#window">
+  <link rel="help" href="https://heycam.github.io/webidl/#Unforgeable">
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<script>
+'use strict';
+
+test(() => {
+  let windowProxy = new Proxy(window, {});
+  name = 'old_name';
+  let descriptor = Object.getOwnPropertyDescriptor(windowProxy, 'name');
+
+  assert_equals(descriptor.get.call(window), 'old_name');
+  descriptor.set.call(window, 'new_name');
+  assert_equals(name, 'new_name');
+  assert_true(descriptor.enumerable);
+  assert_true(descriptor.configurable);
+}, 'Window target, no trap, "name" attribute');
+
+test(() => {
+  let windowProxy = new Proxy(window, {});
+
+  assert_object_equals(
+    Object.getOwnPropertyDescriptor(windowProxy, 'document'),
+    Object.getOwnPropertyDescriptor(window, 'document')
+  );
+}, 'Window target, forwarding trap, [Unforgeable] "document" attribute');
+
+test(() => {
+  let trapResult = {get() {}, set(_val) {}, enumerable: false, configurable: true};
+  let windowProxy = new Proxy(new Proxy(window, {}), {
```

Comment:
Why the double Proxy?

Reply:
It actually the case WebKit was incorrect for. More specifically, `Proxy(window)` or `Proxy(Proxy(document))`.

Apart from [data](https://tc39.es/ecma262/#table-2) and [accessor descriptors](https://tc39.es/ecma262/#table-3), WebKit has a few special descriptors kinds for some DOM magic like throwing cross-origin `location.href` getter. To serialize them in `Object.getOwnPropertyDescriptor` unobservably for user, there was some code that special-based `WindowProxy` and `Proxy` (which is now gone) that used to break for double `Proxy`.